### ISSUE TITLE
When I click the Switch Theme button Twice,the theme Mica is not work

### ISSUE THREAD

**Issue by Smileher** [lepoco/wpfui#927]
### Describe the bug

When I click the Switch Theme button Twice,the theme Mica is not work

### To Reproduce

Switch Dark Theme

### Expected behavior

Mica

### Screenshots

![WeChat截图_20240202132206](https://github.com/lepoco/wpfui/assets/16812755/cbc6289a-94c0-4968-9e4f-6cd8c372d654)
![WeChat截图_20240202132505](https://github.com/lepoco/wpfui/assets/16812755/7420a9fd-b9c8-456d-bd66-569c65fd9622)
![WeChat截图_20240202133049](https://github.com/lepoco/wpfui/assets/16812755/fbd866a6-99e5-4b7a-a076-fa5683c2abe7)

### OS version

Windows 11  23H2

### .NET version

.NET FrameWork 4.8

### WPF-UI NuGet version

3.0.0-preview.13

**Comment by johnvesalius** [comment]
Yeah, I see the same behaviour in 3.0.0. The Mica effect disappears when switching themes and is only applied again when the app is restarted.

**Comment by Muniwedesu** [comment]
By the way, it doesn't work because there was a fix for another issue (ig), which disabled setting background to transparent. 
To get this to work, you need to set your window background to transparent (there's a public method for this, it only needs your window reference)
You can fix that like so:
```cs
{
    ApplicationThemeManager.Apply(appTheme == ApplicationTheme.Light
                    ? ApplicationTheme.Dark
                    : ApplicationTheme.Light);
                    // where 'this' is your window instance, 
                    // winver is client OS build number, but it doesn't seem to break anything 
                    // when you remove background on the earlier versions 
                    // 22621 is windows build number which has mica support, though it may also work fine on some earlier ones
                if (winver > 22621) WindowBackdrop.RemoveBackground(this); 
}
```

**Comment by Smileher** [comment]
> 顺便说一句，它不起作用，因为修复了另一个问题（ig），该问题禁用了将背景设置为透明。要让它工作，你需要将你的窗口背景设置为透明（有一个公共方法，它只需要你的窗口引用）你可以像这样解决这个问题：
> 
> ```cs
> {
>     ApplicationThemeManager.Apply(appTheme == ApplicationTheme.Light
>                     ? ApplicationTheme.Dark
>                     : ApplicationTheme.Light);
>                     // where 'this' is your window instance, 
>                     // winver is client OS build number, but it doesn't seem to break anything 
>                     // when you remove background on the earlier versions 
>                     // 22621 is windows build number which has mica support, though it may also work fine on some earlier ones
>                 if (winver > 22621) WindowBackdrop.RemoveBackground(this); 
> }
> ```

Thank you ，I solved the problem.
`if (Environment.OSVersion.Version.Build > 22621)
WindowBackdrop.RemoveBackground(this);`

**Comment by Jay-o-Way** [comment]
Hey @lepoco, @pomianowski - You have a couple of issues regarding the Mica effect and theme switching. Might be a few duplicates, but would you be so kind to look into these issues? I see a lot of comments in different repositories about them. Thanks!

**Comment by Smileher** [comment]
> By the way, it doesn't work because there was a fix for another issue (ig), which disabled setting background to transparent. To get this to work, you need to set your window background to transparent (there's a public method for this, it only needs your window reference) You can fix that like so:
> 
> ```cs
> {
>     ApplicationThemeManager.Apply(appTheme == ApplicationTheme.Light
>                     ? ApplicationTheme.Dark
>                     : ApplicationTheme.Light);
>                     // where 'this' is your window instance, 
>                     // winver is client OS build number, but it doesn't seem to break anything 
>                     // when you remove background on the earlier versions 
>                     // 22621 is windows build number which has mica support, though it may also work fine on some earlier ones
>                 if (winver > 22621) WindowBackdrop.RemoveBackground(this); 
> }
> ```

Thank you ，I solved the problem.
`if (Environment.OSVersion.Version.Build > 22621) WindowBackdrop.RemoveBackground(this);`

**Comment by Smileher** [comment]
> Hey @lepoco, @pomianowski - You have a couple of issues regarding the Mica effect and theme switching. Might be a few duplicates, but would you be so kind to look into these issues? I see a lot of comments in different repositories about them. Thanks!

Now that I have initially solved this problem, when I change the theme, I need to determine whether the windows system version number is greater than 22621, and when it is greater, I need to remove the background, as mentioned above in @Muniwedesu
